Keyof module import does not have 'default' as an option

<!-- 🚨 STOP 🚨 𝗦𝗧𝗢𝗣 🚨 𝑺𝑻𝑶𝑷 🚨

Half of all issues filed here are duplicates, answered in the FAQ, or not appropriate for the bug tracker. Even if you think you've found a *bug*, please read the FAQ first, especially the Common "Bugs" That Aren't Bugs section!

Please help us by doing the following steps before logging an issue:
  * Search: https://github.com/Microsoft/TypeScript/search?type=Issues
  * Read the FAQ: https://github.com/Microsoft/TypeScript/wiki/FAQ

Please fill in the *entire* template below.
-->

<!-- Please try to reproduce the issue with `typescript@next`. It may have already been fixed. -->
**TypeScript Version:**  3.3.0-dev.20181128, 3.2.0-rc (does not repro in 3.1.6)

<!-- Search terms you tried before logging this (so others can find this issue more easily) -->
**Search Terms:**
keyof module, default, spyOn, jest

**Code**
```ts
// example.ts
export default function add(a: number, b: number) {
    return a + b;
}

//exampleTest.ts
import * as example from '../example';

describe('add', () => {
    it('does something', () => {
        spyOn(example, 'default');
    })
})
```

Uses `@types/jest@23.3.9`, with the following declaration:
```typescript
declare function spyOn<T>(object: T, method: keyof T): jasmine.Spy;
```

**Expected behavior:**
`spyOn(example, 'default')` throws no errors, as the only export from the `example.ts` module is a default export.

**Actual behavior:**
```
src/__tests__/exampleTest.ts:5:24 - error TS2345: Argument of type '"default"' is not assignable to parameter of type '"add"'.
```

**Playground Link:** <!-- A link to a TypeScript Playground "Share" link which demonstrates this behavior --> None; the playground doesn't seem to support multiple modules

**Related Issues:** <!-- Did you find other bugs that looked similar? -->None that I could find.

--- a/src/compiler/checker.ts
+++ b/src/compiler/checker.ts
@@ -9410,8 +9410,13 @@ namespace ts {
             if (!(getDeclarationModifierFlagsFromSymbol(prop) & ModifierFlags.NonPublicAccessibilityModifier)) {
                 let type = getLateBoundSymbol(prop).nameType;
                 if (!type && !isKnownSymbol(prop)) {
-                    const name = prop.valueDeclaration && getNameOfDeclaration(prop.valueDeclaration) as PropertyName;
-                    type = name && getLiteralTypeFromPropertyName(name) || getLiteralType(symbolName(prop));
+                    if (prop.escapedName === InternalSymbolName.Default) {
+                        type = getLiteralType("default");
+                    }
+                    else {
+                        const name = prop.valueDeclaration && getNameOfDeclaration(prop.valueDeclaration) as PropertyName;
+                        type = name && getLiteralTypeFromPropertyName(name) || getLiteralType(symbolName(prop));
+                    }
                 }
                 if (type && type.flags & include) {
                     return type;
@@ -20873,6 +20878,7 @@ namespace ts {
                     if (hasSyntheticDefault) {
                         const memberTable = createSymbolTable();
                         const newSymbol = createSymbol(SymbolFlags.Alias, InternalSymbolName.Default);
+                        newSymbol.nameType = getLiteralType("default");
                         newSymbol.target = resolveSymbol(symbol);
                         memberTable.set(InternalSymbolName.Default, newSymbol);
                         const anonymousSymbol = createSymbol(SymbolFlags.TypeLiteral, InternalSymbolName.Type);

--- a/tests/baselines/reference/keyofModuleObjectHasCorrectKeys.js
+++ b/tests/baselines/reference/keyofModuleObjectHasCorrectKeys.js
@@ -1,0 +1,27 @@
+//// [tests/cases/compiler/keyofModuleObjectHasCorrectKeys.ts] ////
+
+//// [example.ts]
+export default function add(a: number, b: number) {
+    return a + b;
+}
+
+//// [test.ts]
+import * as example from './example';
+
+declare function test<T>(object: T, method: keyof T): void;
+
+test(example, "default");
+
+
+//// [example.js]
+"use strict";
+exports.__esModule = true;
+function add(a, b) {
+    return a + b;
+}
+exports["default"] = add;
+//// [test.js]
+"use strict";
+exports.__esModule = true;
+var example = require("./example");
+test(example, "default");

--- a/tests/baselines/reference/keyofModuleObjectHasCorrectKeys.symbols
+++ b/tests/baselines/reference/keyofModuleObjectHasCorrectKeys.symbols
@@ -1,0 +1,27 @@
+=== tests/cases/compiler/example.ts ===
+export default function add(a: number, b: number) {
+>add : Symbol(add, Decl(example.ts, 0, 0))
+>a : Symbol(a, Decl(example.ts, 0, 28))
+>b : Symbol(b, Decl(example.ts, 0, 38))
+
+    return a + b;
+>a : Symbol(a, Decl(example.ts, 0, 28))
+>b : Symbol(b, Decl(example.ts, 0, 38))
+}
+
+=== tests/cases/compiler/test.ts ===
+import * as example from './example';
+>example : Symbol(example, Decl(test.ts, 0, 6))
+
+declare function test<T>(object: T, method: keyof T): void;
+>test : Symbol(test, Decl(test.ts, 0, 37))
+>T : Symbol(T, Decl(test.ts, 2, 22))
+>object : Symbol(object, Decl(test.ts, 2, 25))
+>T : Symbol(T, Decl(test.ts, 2, 22))
+>method : Symbol(method, Decl(test.ts, 2, 35))
+>T : Symbol(T, Decl(test.ts, 2, 22))
+
+test(example, "default");
+>test : Symbol(test, Decl(test.ts, 0, 37))
+>example : Symbol(example, Decl(test.ts, 0, 6))
+

--- a/tests/baselines/reference/keyofModuleObjectHasCorrectKeys.types
+++ b/tests/baselines/reference/keyofModuleObjectHasCorrectKeys.types
@@ -1,0 +1,27 @@
+=== tests/cases/compiler/example.ts ===
+export default function add(a: number, b: number) {
+>add : (a: number, b: number) => number
+>a : number
+>b : number
+
+    return a + b;
+>a + b : number
+>a : number
+>b : number
+}
+
+=== tests/cases/compiler/test.ts ===
+import * as example from './example';
+>example : typeof example
+
+declare function test<T>(object: T, method: keyof T): void;
+>test : <T>(object: T, method: keyof T) => void
+>object : T
+>method : keyof T
+
+test(example, "default");
+>test(example, "default") : void
+>test : <T>(object: T, method: keyof T) => void
+>example : typeof example
+>"default" : "default"
+

--- a/tests/cases/compiler/keyofModuleObjectHasCorrectKeys.ts
+++ b/tests/cases/compiler/keyofModuleObjectHasCorrectKeys.ts
@@ -1,0 +1,11 @@
+// @filename: example.ts
+export default function add(a: number, b: number) {
+    return a + b;
+}
+
+// @filename: test.ts
+import * as example from './example';
+
+declare function test<T>(object: T, method: keyof T): void;
+
+test(example, "default");

